Assets/Darwin/ColliderArea.cs

```diff
@@ -11,130 +11,187 @@
+        float random = Random.Range(0.0f, 1.0f); // Used to choose if inherited trait will mostly be from parent 1 or 2
 
+        // Inherit most of parent 1's traits
         if(random <= darwinHeredity)
             return ((darwinHeredity + otherHeredity) * darwinTrait + otherHeredity * otherTrait) / (darwinHeredity + otherHeredity * 2);
-        else if(random <= darwinHeredity + otherHeredity)
+        else if(random <= darwinHeredity + otherHeredity) // Inherit most of parent 2's traits
             return ((darwinHeredity + otherHeredity) * otherTrait + darwinHeredity * darwinTrait) / (darwinHeredity * 2 + otherHeredity);
 
+        // Mutation. Create own trait.
         return -1;
     }
 
-    void OnTriggerEnter2D(Collider2D other) // Trigger(used for cookies)
+    /*
+     * Used to check collision with cookie.
+     * Upon collision, reset the Darwin's boredom to 0, increase the Darwin's energy, delete cookie, and spawn a new one somewhere else.
+     * 
+     * @param other: The Collider2D of the cookie the Darwin collided with.
+     */
+    void OnTriggerEnter2D(Collider2D other)
     {
         DarwinTraits darwinTraits = gameObject.GetComponent<DarwinTraits>();
         CookieInstantiator cookieScript = cookieInstantiator.GetComponent<CookieInstantiator>();
 
+        // Hooray! I got a cookie, task successful, reset boredom.
         darwinTraits.Boredom = 0;
 
         if (other.gameObject.CompareTag("Cookie"))
         {
             //Debug.Log("Ate a Cookie!");
             Destroy(other.gameObject);
+            // Increase the Darwin's energy
             darwinTraits.Energy += 5;
             cookieScript.spawnCookie();
         }
     }
 
-    void OnCollisionEnter2D(Collision2D other) // Collision(used for other frogs)
+    /*
+     * Used to check collision with another Darwin.
+     * Upon collision, reset the Darwin's boredom to 0.
+     * If current Darwin and other Darwin are able to breed, generate traits for child to inherit and spawn it.
+     * 
+     * @param other: The Collider2D of the other Darwin the current Darwin collided with.
+     */
+    void OnCollisionEnter2D(Collision2D other)
     {
-        DarwinTraits otherTraits = other.gameObject.GetComponent<DarwinTraits>(); // parent 1
-        DarwinTraits darwinTraits = gameObject.GetComponent<DarwinTraits>(); // parent 2
+        DarwinTraits otherTraits = other.gameObject.GetComponent<DarwinTraits>(); // parent 1's list of traits
+        DarwinTraits darwinTraits = gameObject.GetComponent<DarwinTraits>(); // parent 2's list of traits
 
+        // Hooray! I bumped into another Darwin, task successful, reset boredom.
         darwinTraits.Boredom = 0;
 
         if (other.gameObject.CompareTag("Darwin"))
         {
+            // Are criterias met for both of us to breed?
             if (otherTraits.canBreed() && darwinTraits.canBreed())
             {
+                GameObject newObject; // The gameobject of the new child that will be birthed.
```

Comment:
I see that the line that declared and initialized the newObject GameObject line was changed to 
`GameObject newObject;`
This change looks harmless, but I would consider making any code changes in separate commits to stop bugs from slipping by. I believe your changes intend to be additions or modifications to the documentation, correct? I won't request any changes, but let me know when you see this. Awesome additions to the comments!

Reply:
Yes, I should get into the practice of making separate commits based on the changes I made. Will do in the future!